src/Event/LocalFileStorageListener.php

```diff
@@ -94,12 +94,8 @@ public function afterSave(Event $event) {
 			try {
 				$filename = $this->buildFileName($table, $entity);
 				$entity->path = $this->buildPath($table, $entity);
-
 				$Storage->write($entity['path'] . $filename, file_get_contents($entity['file']['tmp_name']), true);
-				$table->save($entity, array(
-					'validate' => false,
-					'callbacks' => false
-				));
```

Comment:
`callbacks` option is removed in Cake 3, I has to add the `->isNew()` check on the entity to avoid the loop.